src/server/aws_lambda.ts

```diff
@@ -49,21 +53,40 @@ function syncHandler<TEvent = any, TResult = any>(handler: SyncHandler<TEvent, T
     return function wrappedLambdaHandler(event, context, cb) {
         hb.__setStore(AsyncStore)
         AsyncStore.run({context: {}, breadcrumbs: []}, () => {
+            const timeoutHandler = setupTimeoutWarning(hb, context)
             try {
                 handler(event, context, (error, result) => {
+                    clearTimeout(timeoutHandler)
                     if (error) {
                         return reportToHoneybadger(hb, error, cb)
                     }
 
                     cb(null, result)
                 });
             } catch (err) {
+                clearTimeout(timeoutHandler)
                 reportToHoneybadger(hb, err, cb)
             }
         })
     }
 }
 
+function shouldReportTimeoutWarning(hb: Honeybadger, context: Context): boolean {
+    return typeof context.getRemainingTimeInMillis === 'function' && !!((hb.config as ServerlessConfig).reportTimeoutWarning)
+}
+
+function setupTimeoutWarning(hb: Honeybadger, context: Context) {
+    if (!shouldReportTimeoutWarning(hb, context)) {
+        return
+    }
+
+    const delay = context.getRemainingTimeInMillis() - ((hb.config as ServerlessConfig).timeoutWarningThresholdMs)
+    return setTimeout(() => {
+        hb.notify(`${context.functionName}[${context.functionVersion}] may timeout`)
```

Comment:
Maybe "may have timed out?" When they see the notification, it'll be in the past.

Reply:
Good point!